python/lsst/pipe/tasks/make_direct_warp.py

```diff
@@ -697,58 +736,103 @@ def _apply_all_calibrations(
+            # This could get trigged only if runQuantum is skipped and run is
+            # called directly.
+            raise RuntimeError(
+                f"doRevertOldBackground is False, but {detector_inputs.data_id} has a background_revert."
+            )
 
-        if self.config.useVisitSummaryPsf:
-            detector = exp.info.getDetector().getId()
+        if visit_summary is not None:
+            detector = detector_inputs.exposure.info.getDetector().getId()
             row = visit_summary.find(detector)
 
             if row is None:
-                raise RuntimeError(f"Unexpectedly incomplete visit_summary: {detector=} is missing.")
+                self.log.info(
+                    "Unexpectedly incomplete visit_summary: detector = %s is missing. Skipping it.",
+                    detector,
+                )
+                return False
 
             if photo_calib := row.getPhotoCalib():
-                exp.setPhotoCalib(photo_calib)
+                detector_inputs.exposure.setPhotoCalib(photo_calib)
             else:
-                logger.warning(
-                    "No photometric calibration found in visit summary for detector = %s.",
+                self.log.info(
+                    "No photometric calibration found in visit summary for detector = %s. Skipping it.",
                     detector,
                 )
+                return False
 
             if wcs := row.getWcs():
-                exp.setWcs(wcs)
+                detector_inputs.exposure.setWcs(wcs)
             else:
-                logger.warning("No WCS found in visit summary for detector = %s.", detector)
+                self.log.info("No WCS found in visit summary for detector = %s. Skipping it.", detector)
+                return False
 
-            if psf := row.getPsf():
-                exp.setPsf(psf)
-            else:
-                logger.warning("No PSF found in visit summary for detector = %s.", detector)
+            if self.config.useVisitSummaryPsf:
+                if psf := row.getPsf():
+                    detector_inputs.exposure.setPsf(psf)
+                else:
+                    self.log.info("No PSF found in visit summary for detector = %s. Skipping it.", detector)
+                    return False
 
-            if apcorr_map := row.getApCorrMap():
-                exp.setApCorrMap(apcorr_map)
-            else:
-                logger.warning(
-                    "No aperture correction map found in visit summary for detector = %s.",
-                    detector,
+                if apcorr_map := row.getApCorrMap():
+                    detector_inputs.exposure.setApCorrMap(apcorr_map)
+                else:
+                    self.log.info(
+                        "No aperture correction map found in visit summary for detector = %s. Skipping it",
+                        detector,
+                    )
+                    return False
+
+            elif self.config.useVisitSummaryPsf:
```

Comment:
This is an `elif` on the same condition as the original `if`.  Is it supposed to be an `else`?  I'm not actually sure it should exist, because if `useVisitSummaryPsf` is `False`, we should just not try to get a PSF from the visit summary.